src/terminal_ui.cc

```diff
@@ -1574,9 +1574,14 @@ void TerminalUI::set_ui_options(const Options& options)
 
     m_padding_char = find("terminal_padding_char").map([](StringView s) { return s.column_length() < 1 ? ' ' : s[0_char]; }).value_or(Codepoint{'~'});
     m_padding_fill = find("terminal_padding_fill").map(to_bool).value_or(false);
-
-    // Emit cursor show sequence
-    write(STDOUT_FILENO, "\033[?25h");
+    
+    bool new_cursor_native = find("terminal_cursor_native").map(to_bool).value_or(false);
+    if (new_cursor_native != m_terminal_cursor_native)
+    {
+        m_terminal_cursor_native = new_cursor_native;
+        // Emit cursor visibility command when the option changes
```

Comment:
that comment does not add much value, it seems to just describe what the code obviously does.